docs/meshstack.configuration.md

```diff
@@ -134,11 +135,203 @@ When a user is invited to a customer there are several configurations in order t
+            UserByUsernameLookupStrategy or UserByMailLookupStrategy. The UserByMailLookupStrategy
+            uses the users euid and uses it as an E-Mail address for AAD lookup. The
+            UserByUsernameLookupStrategy assumes the euid is an userPrincipalName for AAD lookup.
+    -}
+      { aad-tenant : Text
+      , client-id : Text
+      , client-secret : Secret.Type
+      , user-lookup-strategy : AzureLookupStrategy
+      }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureCreds
+    = { aad-tenant = "devmeshcloud.onmicrosoft.com"
+      , client-id = "f112f31-248a-4461-1269-0f13164acb95"
+      , client-secret = Secret.fromAnsible "client_secret"
+      , user-lookup-strategy =
+          AzureLookupStrategy.UserByMailLookupStrategy
+      }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+If this initial config if present you can decide to setup the following optional user identity steps:
+
+<!--snippet:mesh.meshfed.identity.azure.guest#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let AzureGuestDetection =
+    {-
+      When adding/inviting a new user to a customer a custom attribute property from the AAD data
+      can be used to determine if it is a guest user.
+      Attention: This check is only performed on the first attempt when a user is added/invited
+      to a customer. If this check is configured after some users were initially added to a
+      customer they are not detected as guest users anymore.
+
+        guest-property:
+            The AAD's custom attribute which is checked against the guest-value.
+
+        guest-value:
+            If the AAD custom attribute matches this value the user is considered to be a guest.
+    -}
+      { guest-property : Text, guest-value : Text }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureGuestDetection
+    = { guest-property = "guest-property", guest-value = "is-guest" }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+<!--snippet:mesh.meshfed.identity.azure.euid#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let AzureEuidExtensionSchema =
+    {-
+      When adding/inviting a new user to a customer a custom attribute property from the users AAD
+      extension schema can be used to fill in his euid.
+
+      Attention: This check is only performed on the first attempt when a user is added/invited
+      to a customer. If this check is configured after some users were initially added to a
+      customer their euid's are not udpated anymore.
+
+        euidExtensionSchemaProperty:
+            The name of the extension schema property which is used as EUID value.
+    -}
+      { euidExtensionSchemaProperty : Text
+      , euidExtensionSchemaIdentifier : Text
+      }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureEuidExtensionSchema
+    = { euidExtensionSchemaProperty = "euid-schema-property"
+      , euidExtensionSchemaIdentifier = "euid-schema-identfifier"
```

Comment:
d: If we only support accessing properties that are inside schema extensions, this should be made clear in docs.

Reply:
I made it clear in the DHALL comments (which are also in the docs) that normal properties also work out for that cause.